polyffusion/prepare_data.py

```diff
@@ -0,0 +1,104 @@
+import os
+import muspy
+from tqdm import tqdm
+import numpy as np
+from data.midi_to_data import *
+from data.dataset import PianoOrchDataset
+from argparse import ArgumentParser
+
+def force_length(music : muspy.music, bars = 8):
+    for track in music.tracks:
+        timesteps = track.get_end_time()
+        old_bars = (timesteps + 15) // 16
+        div = bars // old_bars
+        for i in range(1, div):
+            tmp = track.deepcopy()
+            tmp.adjust_time(lambda x : x + i * timesteps)
+            track.notes.extend(tmp.notes)
```

Comment:
Sorry for the lack of documentation. This function is for the case when the user wants to train short loops which are usually either 4-bars or 8-bars in length per MIDI. For the ones that are 4-bars, we use this function to loop it twice. Note that this `force_length` function is disabled by default when preparing data.

It is true that this would not function as intended if a MIDI has 1 track that's >= 8 bars but some other tracks < 8 bars, as it would loop the tracks < 8 bars. To address this, I implemented a simple check in the newest pull request.